x/rollup/keeper/deposits.go

```diff
@@ -109,3 +184,60 @@ func (k *Keeper) mintETH(ctx sdk.Context, addr sdk.AccAddress, amount sdkmath.In
 
 	return &mintEvent, nil
 }
+
+// mintERC20 mints a bridged ERC-20 token to an account and returns the associated event.
+func (k *Keeper) mintERC20(
+	ctx sdk.Context, //nolint:gocritic // hugeParam
+	userAddr sdk.AccAddress,
+	erc20addr string,
+	amount sdkmath.Int,
+) (*sdk.Event, error) {
+	// use the "erc20/{l1erc20addr}" format for the coin denom
+	coin := sdk.NewCoin("erc20/"+erc20addr[2:], amount)
+	if err := k.bankkeeper.MintCoins(ctx, types.ModuleName, sdk.NewCoins(coin)); err != nil {
+		return nil, fmt.Errorf("failed to mint ERC-20 deposit coins to the rollup module: %v", err)
+	}
+	if err := k.bankkeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, userAddr, sdk.NewCoins(coin)); err != nil {
+		return nil, fmt.Errorf("failed to send ERC-20 deposit coins from rollup module to user account %v: %v", userAddr, err)
+	}
+
+	mintEvent := sdk.NewEvent(
+		types.EventTypeMintERC20,
+		sdk.NewAttribute(types.AttributeKeyL1DepositTxType, types.L1UserDepositTxType),
+		sdk.NewAttribute(types.AttributeKeyToCosmosAddress, userAddr.String()),
+		sdk.NewAttribute(types.AttributeKeyERC20Address, erc20addr),
+		sdk.NewAttribute(types.AttributeKeyValue, hexutil.Encode(amount.BigInt().Bytes())),
+	)
+
+	return &mintEvent, nil
+}
+
+// TODO: should this belong in the utils package?
+// unpackInputsIntoInterface unpacks the input data of a function call into an interface. This function behaves
+// similarly to the geth abi UnpackIntoInterface function but unpacks method inputs instead of outputs.
+func unpackInputsIntoInterface(contractABI *abi.ABI, methodName string, inputData []byte, outputInterface interface{}) error {
```

Comment:
TODO: Should we just swallow all errors here? Or should we only swallow errors for unrecognized L1StandardBridge messages?